hydromt_sfincs/workflows/merge.py

```diff
@@ -256,6 +256,75 @@ def merge_dataarrays(
     return da_out
 
 
+# Merge data for Curvenumber
+def curvenumber_recovery_determination(
+    da_landuse, da_HSG, da_Ksat, df_map, da_mask_block
+):
+    """Setup model the Soil Conservation Service (SCS) Curve Number (CN) files.
+    More information see http://new.streamstech.com/wp-content/uploads/2018/07/SWMM-Reference-Manual-Part-I-Hydrology-1.pdf
+
+    Parameters
+    ----------
+    dataset_landuse : filename (or Path) of gridded data with land use classes (e.g. NLCD)
+    dataset_HSG     : filename (or Path) of gridded data with hydrologic soil group classes (HSG)
+    dataset_Ksat    : filename (or Path) of gridded data with saturated hydraulic conductivity (Ksat)
+    reclass_table   : mapping table that related landuse and HSG to each other (matrix; not list)
+    """
+    # Started
+    print(" processing curve numbers")
+    da_smax = xr.full_like(da_mask_block, -9999, dtype=np.float32)
+    da_kr = xr.full_like(da_mask_block, -9999, dtype=np.float32)
+
+    # Interpolate soil type to landuse
+    da_HSG_to_landuse = da_HSG.raster.reproject_like(
+        da_landuse, method="nearest"
+    ).load()
+
+    # Curve numbers to grid: go over NLCD classes and HSG classes
+    da_CN = xr.full_like(da_landuse, np.NaN, dtype=np.float32)
+    for i in range(len(df_map._stat_axis)):
```

Comment:
I think you could replace `len(df_map. _stat_axis)` with `df_map.index.size` but not entirely sure because I don't know the _stat_axis attribute (and neither does google..). Generally, attributes and methods starting with an underscore are part of the internal API of a package an may change without warning. If possible I prefer to not use these.

Reply:
OK, good to know. Just changed this.